Content.Server/Remotes/DoorRemoteSystem.cs

```diff
@@ -0,0 +1,92 @@
+        {
+            switch (component.Mode)
+            {
+                case DoorRemoteComponent.OperatingMode.OpenClose:
+                    component.Mode = DoorRemoteComponent.OperatingMode.ToggleBolts;
+                    _popupSystem.PopupEntity(Loc.GetString("door-remote-switch-state-toggle-bolts"), args.User, Filter.Entities(args.User));
+                    break;
+                case DoorRemoteComponent.OperatingMode.ToggleBolts:
+                    component.Mode = DoorRemoteComponent.OperatingMode.OpenClose; // TODO: Swítch to ToggleEmergencyAcces when EA is implemented
+                    _popupSystem.PopupEntity(Loc.GetString("door-remote-switch-state-open-close"), args.User, Filter.Entities(args.User)); // TODO: See the above comment
+                    break;
+            /*
+                case DoorRemoteComponent.OperatingMode.ToggleEmergencyAccess:
+                    component.Mode = DoorRemoteComponent.OperatingMode.OpenClose;
+                    _popupSystem.PopupEntity(Loc.GetString("door-remote-switch-state-open-close"), args.User, Filter.Entities(args.User));
+                    break;
+            */
+            }
+        }
+        private void OnAfterInteract(EntityUid uid, DoorRemoteComponent component, AfterInteractEvent args)
+        {
+            if (args.Handled
+                || args.Target == null
+                || !EntityManager.TryGetComponent<DoorComponent>(args.Target, out var doorComponent) // If it isn't a door we don't use it
+                || !EntityManager.HasComponent<AccessReaderComponent>(args.Target) // Remotes do not work on doors without access requirements
+                || !EntityManager.TryGetComponent<AirlockComponent>(args.Target, out var airlockComponent) // Remotes only work on airlocks
+                || !_interactionSystem.InRangeUnobstructed(args.User, doorComponent.Owner, -1f, CollisionGroup.Opaque))
+
+            {
+                return;
+            }
+
+            args.Handled = true;
+            
+            if (component.Mode == DoorRemoteComponent.OperatingMode.OpenClose)
+            {
+                _sharedDoorSystem.TryToggleDoor(doorComponent.Owner, user: args.Used);
+            }
+
+            if (component.Mode == DoorRemoteComponent.OperatingMode.ToggleBolts
+                && airlockComponent.IsPowered())
+            {
+                if (_doorSystem.HasAccess(args.Used))
+                {
+                    if(airlockComponent.IsBolted())
+                    {
+                        airlockComponent.SetBoltsWithAudio(false);
+                    }
+                    else
+                    {
+                        airlockComponent.SetBoltsWithAudio(true);
+                    }
+                }
+                else
+                {
+                    _sharedDoorSystem.Deny(airlockComponent.Owner, user: args.User);
```

Comment:
FIY door denying only works when the door is currently closed, I never though about doors needing to do a "deny" when currently open.

IF you just want have general feedback even for when people try to bolt a door open, maybe just directly play the sound if its currently open, so something like
```cs
if (doorComponent.State != DoorState.Open)
    _sharedDoorSystem.Deny(args.Target.Value)
else if (doorComponent.DenySound != null)
    SoundSystem.Play(Filter.Pvs(args.Target.Value), door.DenySound.GetSound(), args.Target.Value);
```